graphql_compiler/compiler/expressions.py

```diff
@@ -928,11 +937,22 @@ def to_sql(self, dialect: Any, aliases: AliasesDictType, current_alias: AliasTyp
             # PostgreSQL
             # coalesce to an empty array of the corresponding type
             inner_type = strip_non_null_from_type(self.field_type.of_type)
-            if is_same_type(GraphQLInt, inner_type):
-                sql_array_type = "INT"
-            elif is_same_type(GraphQLString, inner_type):
-                sql_array_type = "VARCHAR"
-            else:
+            graphql_type_to_sql_array_type_dict = {
+                GraphQLInt: "INT",
+                GraphQLString: "VARCHAR",
+                GraphQLID: "UUID",
+                GraphQLBoolean: "BOOL",
+                GraphQLFloat: "FLOAT",
+                GraphQLDate: "DATE",
+                GraphQLDateTime: "TIMESTAMP",
+                GraphQLDecimal: "FLOAT",
+            }
+            sql_array_type = None
+            for graphql_type, type_name in graphql_type_to_sql_array_type_dict.items():
+                if is_same_type(graphql_type, inner_type):
+                    sql_array_type = type_name
+                    break
```

Comment:
This function is getting pretty long. How about pulling this type-matching logic into its own helper function, so that instead of `break` here you could have `return type_name`?